RecoMET/METFilters/plugins/BadParticleFilter.cc

```diff
@@ -196,5 +219,24 @@ bool BadParticleFilter::filter(edm::StreamID iID, edm::Event& iEvent, const edm:
   return taggingMode_ || pass;
 }
 
+void BadParticleFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  // BadPFMuonFilter
+  edm::ParameterSetDescription desc;
+  desc.add<double>("innerTrackRelErr", 1.0);
+  desc.add<double>("minDzBestTrack", -1.0);
+  desc.add<edm::InputTag>("PFCandidates", edm::InputTag("particleFlow"));
+  desc.add<std::string>("filterType", "BadPFMuon");
+  desc.add<double>("segmentCompatibility", 0.3);
+  desc.add<double>("minMuonPt", 100);
+  desc.add<int>("algo", 14);
+  desc.add<bool>("taggingMode", false);
+  desc.add<edm::InputTag>("vtx", edm::InputTag("offlinePrimaryVertices"));
+  desc.add<double>("minMuonTrackRelErr", 2.0);
+  desc.add<double>("maxDR", 0.001);
+  desc.add<edm::InputTag>("muons", edm::InputTag("muons"));
+  desc.add<double>("minPtDiffRel", 0.0);
+  descriptions.add("BadPFMuonFilter", desc);
```

Comment:
As such, two identical instances of the 'BadParticleFilter' are created by this ```fillDescriptions()```:
```
 BadPFMuonFilter_cfi.py
 badParticleFilter_cfi.py
```

In the configurations as updated below, other instances are cloned-and-modified from the BadPFMuonFilter. I think that badParticleFilter is  a more generic name for the filter to clone from.

Therefore, I would suggest the following:
- Remove the line
```
  descriptions.add("BadPFMuonFilter", desc);
```
so that only one instance of BadParticleFilter is automatically created
- Make all configurations below to clone from ```badParticleFilter``` instead from ```BadPFMuonFilter```
- Re-instantiate a ```BadPFMuonFilter_cfi.py``` in ```RecoMET/METFilters/python``` that simply clones from the generic ```badParticleFilter```, without any modification of the default settings

@slava77 : would this still go in the spirit of your comment?

Reply:
Hi @perrotta , @slava77's suggestion was to clone from "badParticleFilter" but I was not able to run successfully using badParticleFilter in my local area. 
BadChargedCandidateFilter = _mod.badParticleFilter.clone(
    filterType  ="BadChargedCandidate",
    maxDR = 0.00001,
    minPtDiffRel = 0.00001
)

Reply:
If you include
```
import  RecoMET.METFilters.badParticleFilter_cfi as _mod
```
instead of
```
import  RecoMET.METFilters.BadPFMuonFilter_cfi as _mod
```
it should work

Reply:
> @slava77 : would this still go in the spirit of your comment?

I'm not aiming for a perfect implementation at this point.
In my previous comments I did ask to add badParticleFilter_cfi to BadPFMuonFilter_cfi so that for a non-muon-related instances we can clone formally from badParticleFilter_cfi.

Reply:
Hi @slava77 , @perrotta 
I changed  cfi's to 'import  RecoMET.METFilters.badParticleFilter_cfi' as _mod and the battery tests ran fine in my local area.

Shall I create BadPFMuonFilter_cfi.py file and clone badParticleFilter_cfi with default settings and remove this 
  'descriptions.add("BadPFMuonFilter", desc);' in my next commit?

Please let me know